File: inetpub/wwwroot/AutohausPortal/Admin/PageElements/GridNavigation.ascx.vb
Partial Public Class GridNavigation
    Inherits System.Web.UI.UserControl

    Protected WithEvents mGridView As GridView
    Protected WithEvents mDataGrid As DataGrid
    Public Event PageSizeChanged As PageSize_Changed
    Public Event PagerChanged As Pager_Changed
    Delegate Sub PageSize_Changed()
    Delegate Sub Pager_Changed(ByVal PageIndex As Integer)
    Private PageCount As Integer
    Private PagerStart As Integer
    Private m_PagerSize As Int32

    Public Property PagerSize As Integer
        Get
            Return m_PagerSize
        End Get
        Set(ByVal Value As Integer)
            m_PagerSize = Value
        End Set


    End Property
    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Not IsPostBack Then
            fillPageSizeDropDown()
        End If

    End Sub

    Public Sub setGridElment(ByRef GridElement As GridView)
        mGridView = GridElement
    End Sub

    Public Sub setGridElment(ByRef GridElement As DataGrid)
        mDataGrid = GridElement
    End Sub

    Public Sub setGridTitle(ByVal Title As String)
        lbltitle.Text = Title
        If Title.Trim.Length > 0 Then
            lbltitle.Visible = True
        End If

    End Sub

    Public Sub ProofPager()
        If Not mGridView Is Nothing Then

            If mGridView.PageCount = 1 Then
                lbtnNext.Visible = False
                lbtnPrevious.Visible = False
                Repeater1.Visible = False
                Exit Sub
            Else
                lbtnNext.Visible = True
                lbtnPrevious.Visible = True
                Repeater1.Visible = True
            End If

            If mGridView.PageCount - 1 = mGridView.PageIndex Then
                lbtnNext.Visible = False
                lbtnPrevious.Visible = True
            Else
                lbtnNext.Visible = True
            End If

            If mGridView.PageIndex > 0 Then
                lbtnPrevious.Visible = True
            ElseIf mGridView.PageIndex = 0 Then
                lbtnPrevious.Visible = False
                lbtnNext.Visible = True
            End If
        End If

        If Not mDataGrid Is Nothing Then

            If mDataGrid.PageCount = 1 Then
                lbtnNext.Visible = False
                lbtnPrevious.Visible = False
                Repeater1.Visible = False
                Exit Sub
            Else
                lbtnNext.Visible = True
                lbtnPrevious.Visible = True
                Repeater1.Visible = True
            End If


            If mDataGrid.PageCount - 1 = mDataGrid.CurrentPageIndex Then
                lbtnNext.Visible = False
                lbtnPrevious.Visible = True
            Else
                lbtnNext.Visible = True
            End If
            If mDataGrid.CurrentPageIndex > 0 Then
                lbtnPrevious.Visible = True

            ElseIf mDataGrid.CurrentPageIndex = 0 Then
                lbtnPrevious.Visible = False
                lbtnNext.Visible = True
            End If

        End If

    End Sub

    Private Sub fillPageSizeDropDown()
        ddlPageSize.Items.Add("10")
        ddlPageSize.Items.Add("20")
        ddlPageSize.Items.Add("50")
        ddlPageSize.Items.Add("100")
        ddlPageSize.Items.Add("200")
        ddlPageSize.SelectedIndex = m_PagerSize
    End Sub

    Private Sub GridView1_DataBound(ByVal sender As Object, ByVal e As System.EventArgs) Handles mGridView.DataBound
        ' Retrieve the pager row.

        Dim i As Integer
        Dim LinkTable As New DataTable

        LinkTable.Columns.Add("Index", System.Type.GetType("System.String"))
        LinkTable.Columns.Add("Page", System.Type.GetType("System.String"))
        Dim RepRow As DataRow


        If mGridView.PageCount > 9 Then
            lbtnNext10.Visible = True
        Else
            lbtnNext10.Visible = False
        End If
        If ((mGridView.PageIndex).ToString.Length > 1) Then
            Dim sPage As String = Left((mGridView.PageIndex).ToString, 1)
            sPage += "0"
            PagerStart = CInt(sPage)
            PageCount = CInt(sPage) + 9
            lbtnPrevious10.Visible = True
        Else
            PagerStart = 0
            If mGridView.PageCount > 9 Then
                PageCount = 9
            Else
                PageCount = mGridView.PageCount - 1
            End If

            lbtnPrevious10.Visible = False
        End If


        For i = PagerStart To PageCount

            Dim pageNumber As Integer = i
            RepRow = LinkTable.NewRow
            RepRow("Index") = pageNumber
            RepRow("Page") = pageNumber + 1
            LinkTable.Rows.Add(RepRow)

        Next i


        If LinkTable.Rows.Count > 0 Then
            Repeater1.DataSource = LinkTable
            Repeater1.DataBind()

            For i = 0 To LinkTable.Rows.Count - 1
                If i + mGridView.PageIndex = mGridView.PageIndex Then
                    Dim lbtn As LinkButton
                    If mGridView.PageIndex = 0 Then
                        lbtn = CType(Repeater1.Controls(0).Controls(0).FindControl("lbtnPage"), LinkButton)
                    Else
                        lbtn = CType(Repeater1.Controls(mGridView.PageIndex - PagerStart).Controls(0).FindControl("lbtnPage"), LinkButton)
                    End If

                    lbtn.CssClass = "active"

                End If
            Next
        End If

        ProofPager()
    End Sub

    Private Sub DataGrid1_DataBound(ByVal sender As Object, ByVal e As System.EventArgs) Handles mDataGrid.ItemDataBound

        Dim i As Integer
        Dim LinkTable As New DataTable

        LinkTable.Columns.Add("Index", System.Type.GetType("System.String"))
        LinkTable.Columns.Add("Page", System.Type.GetType("System.String"))
        Dim RepRow As DataRow


        If mDataGrid.PageCount > 9 Then
            lbtnNext10.Visible = True
        Else
            lbtnNext10.Visible = False
        End If
        If ((mDataGrid.CurrentPageIndex).ToString.Length > 1) Then
            Dim sPage As String = Left((mDataGrid.CurrentPageIndex).ToString, 1)
            sPage += "0"
            PagerStart = CInt(sPage)
            PageCount = CInt(sPage) + 9
            lbtnPrevious10.Visible = True
        Else
            PagerStart = 0
            If mDataGrid.PageCount > 9 Then
                PageCount = 9
            Else
                PageCount = mDataGrid.PageCount - 1
            End If

            lbtnPrevious10.Visible = False
        End If


        For i = PagerStart To PageCount

            Dim pageNumber As Integer = i
            RepRow = LinkTable.NewRow
            RepRow("Index") = pageNumber
            RepRow("Page") = pageNumber + 1
            LinkTable.Rows.Add(RepRow)

        Next i


        If LinkTable.Rows.Count > 0 Then
            Repeater1.DataSource = LinkTable
            Repeater1.DataBind()

            For i = 0 To LinkTable.Rows.Count - 1
                If i + mDataGrid.CurrentPageIndex = mDataGrid.CurrentPageIndex Then
                    Dim lbtn As LinkButton
                    If mDataGrid.CurrentPageIndex = 0 Then
                        lbtn = CType(Repeater1.Controls(0).Controls(0).FindControl("lbtnPage"), LinkButton)
                    Else
                        lbtn = CType(Repeater1.Controls(mDataGrid.CurrentPageIndex - PagerStart).Controls(0).FindControl("lbtnPage"), LinkButton)
                    End If

                    lbtn.CssClass = "active"

                End If
            Next
        End If

        ProofPager()

    End Sub


    Private Sub ddlPageSize_SelectedIndexChanged1(ByVal sender As Object, ByVal e As System.EventArgs) Handles ddlPageSize.SelectedIndexChanged
        Select Case False
            Case mDataGrid Is Nothing
                mDataGrid.PageSize = ddlPageSize.SelectedValue
            Case mGridView Is Nothing
                mGridView.PageSize = ddlPageSize.SelectedValue
            Case Else
                Throw New Exception("GridNavigation: noch nicht initialisiert")
        End Select
        RaiseEvent PageSizeChanged()
    End Sub


    Public Sub lbtnPage_PageIndexChanging(ByVal sender As Object, ByVal e As System.EventArgs)
        RaiseEvent PagerChanged(CInt(sender.CommandArgument))
    End Sub

    Private Sub lbtnPrevious_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles lbtnPrevious.Click

        Dim tmpIntPageintex As Integer
        Select Case False
            Case mDataGrid Is Nothing
                tmpIntPageintex = mDataGrid.CurrentPageIndex - 1
            Case mGridView Is Nothing
                tmpIntPageintex = mGridView.PageIndex - 1
            Case Else
                Throw New Exception("GridNavigation: noch nicht initialisiert")
        End Select
        RaiseEvent PagerChanged(tmpIntPageintex)
    End Sub

    Private Sub lbtnNext_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles lbtnNext.Click
        Dim tmpIntPageintex As Integer
        Select Case False
            Case mDataGrid Is Nothing
                tmpIntPageintex = mDataGrid.CurrentPageIndex + 1

            Case mGridView Is Nothing
                tmpIntPageintex = mGridView.PageIndex + 1
            Case Else
                Throw New Exception("GridNavigation: noch nicht initialisiert")
        End Select
        RaiseEvent PagerChanged(tmpIntPageintex)
    End Sub


    Private Sub Page_PreRender(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.PreRender
        ProofPager()

        lblAnzahl.Visible = True
        Select Case False
            Case mDataGrid Is Nothing
                If TypeOf mDataGrid.DataSource Is DataView Then
                    lblAnzahl.Text = "Gesamtanzahl: " & CType(mDataGrid.DataSource, DataView).Count
                ElseIf TypeOf mDataGrid.DataSource Is DataTable Then
                    lblAnzahl.Text = "Gesamtanzahl: " & CType(mDataGrid.DataSource, DataTable).Rows.Count
                Else
                    lblAnzahl.Visible = False
                End If
            Case mGridView Is Nothing
                If TypeOf mGridView.DataSource Is DataView Then
                    lblAnzahl.Text = "Gesamtanzahl: " & CType(mGridView.DataSource, DataView).Count
                ElseIf TypeOf mGridView.DataSource Is DataTable Then
                    lblAnzahl.Text = "Gesamtanzahl: " & CType(mGridView.DataSource, DataTable).Rows.Count
                Else
                    lblAnzahl.Visible = False
                End If
            Case Else
                Throw New Exception("GridNavigation: noch nicht initialisiert")
        End Select

    End Sub

    Protected Sub lbtnNext10_Click(ByVal sender As Object, ByVal e As EventArgs) Handles lbtnNext10.Click
        Dim tmpIntPageintex As Integer
        Select Case False
            Case mDataGrid Is Nothing
                tmpIntPageintex = mDataGrid.CurrentPageIndex + 9
                If ((tmpIntPageintex + 1).ToString.Length > 1) Then
                    Dim sPage As String = Left((tmpIntPageintex + 1).ToString, 1)
                    sPage += "0"
                    PagerStart = CInt(sPage)
                End If
                tmpIntPageintex = PagerStart
            Case mGridView Is Nothing
                tmpIntPageintex = mGridView.PageIndex + 9
                If ((tmpIntPageintex + 1).ToString.Length > 1) Then
                    Dim sPage As String = Left((tmpIntPageintex + 1).ToString, 1)
                    sPage += "0"
                    PagerStart = CInt(sPage)
                End If
                tmpIntPageintex = PagerStart
            Case Else
                Throw New Exception("GridNavigation: noch nicht initialisiert")
        End Select
        RaiseEvent PagerChanged(tmpIntPageintex)
    End Sub

    Protected Sub lbtnPrevious10_Click(ByVal sender As Object, ByVal e As EventArgs) Handles lbtnPrevious10.Click
        Dim tmpIntPageintex As Integer
        Select Case False
            Case mDataGrid Is Nothing
                tmpIntPageintex = mDataGrid.CurrentPageIndex - 9
                If ((tmpIntPageintex).ToString.Length > 1) Then
                    Dim sPage As String = Left((tmpIntPageintex).ToString, 1)
                    sPage += "0"
                    PagerStart = CInt(sPage)
                End If
                tmpIntPageintex = PagerStart
            Case mGridView Is Nothing
                tmpIntPageintex = mGridView.PageIndex - 9
                If ((tmpIntPageintex).ToString.Length > 1) Then
                    Dim sPage As String = Left((tmpIntPageintex).ToString, 1)
                    sPage += "0"
                    PagerStart = CInt(sPage)
                End If
                tmpIntPageintex = PagerStart
            Case Else
                Throw New Exception("GridNavigation: noch nicht initialisiert")
        End Select
        RaiseEvent PagerChanged(tmpIntPageintex)
    End Sub
End Class

' ************************************************
' $History: GridNavigation.ascx.vb $
' 
' *****************  Version 1  *****************
' User: Rudolpho     Date: 11.03.11   Time: 13:59
' Created in $/CKPortalZLD/PortalZLD/PageElements
' 
' *****************  Version 13  *****************
' User: Rudolpho     Date: 24.08.10   Time: 16:21
' Updated in $/CKAG2/Services/PageElements
' 
' *****************  Version 12  *****************
' User: Dittbernerc  Date: 29.07.09   Time: 11:48
' Updated in $/CKAG2/Services/PageElements
' 
' *****************  Version 11  *****************
' User: Rudolpho     Date: 5.05.09    Time: 17:16
' Updated in $/CKAG2/Services/PageElements
' 
' *****************  Version 10  *****************
' User: Rudolpho     Date: 28.04.09   Time: 15:45
' Updated in $/CKAG2/Services/PageElements
' 
' *****************  Version 9  *****************
' User: Rudolpho     Date: 23.04.09   Time: 13:36
' Updated in $/CKAG2/Services/PageElements
' 
' *****************  Version 8  *****************
' User: Rudolpho     Date: 14.04.09   Time: 17:19
' Updated in $/CKAG2/Services/PageElements
' 
' *****************  Version 7  *****************
' User: Jungj        Date: 14.04.09   Time: 16:52
' Updated in $/CKAG2/Services/PageElements
' 
' *****************  Version 6  *****************
' User: Jungj        Date: 14.04.09   Time: 16:48
' Updated in $/CKAG2/Services/PageElements
' 
' ************************************************

File: inetpub/wwwroot/Kantine/PageElements/GridNavigation.ascx.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Web;
using System.Web.UI.WebControls;

namespace Kantine
{
	public partial class GridNavigation : System.Web.UI.UserControl
	{
		protected GridView mGridView;
		protected DataGrid mDataGrid;

        public delegate void PageSize_ChangedEventHandler();
        public delegate void Pager_ChangedEventHandler(object sender, int pageindex);

        public event PageSize_ChangedEventHandler PageSizeChanged;
        public event Pager_ChangedEventHandler PagerChanged;

		private int PageCount;
		private int PagerStart;


        protected GridNavigation()
        {
             
        }

		protected void Page_Load(object sender, EventArgs e)
		{
            if (mDataGrid != null)
            {
                mDataGrid.ItemDataBound += new DataGridItemEventHandler(DataGrid_DataBinding);               
            }
            else if (mGridView != null)
            {
                mGridView.DataBound += new EventHandler(GridView_DataBinding);
            }
			if (!IsPostBack)
			{
			    fillPageSizeDropDown();
                ProofPager();
			}
		}

        protected void GridView_DataBinding(object sender,  EventArgs e)
		{
			ProofPager(); 
		}

        protected void DataGrid_DataBinding(object sender, DataGridItemEventArgs e)
		{
            if (e.Item.ItemIndex == -1)
            {
                ProofPager();
            }
		}

		public void setGridElement(ref GridView GridElement)
		{
			mGridView = GridElement;			
		}

		public void setGridElement(ref DataGrid GridElement)
		{
			mDataGrid = GridElement;			
		}

		public void setGridTitle(String Title)
		{
			lbltitle.Text = Title;
			if (Title.Trim().Length > 0)
			{
				lbltitle.Visible = true;
			}
		}

		private void fillPageSizeDropDown()
		{
			ddlPageSize.Items.Add("10");
			ddlPageSize.Items.Add("20");
			ddlPageSize.Items.Add("50");
			ddlPageSize.Items.Add("100");
			ddlPageSize.Items.Add("200");
			ddlPageSize.Items.Add("500");
			ddlPageSize.SelectedIndex = 1;
		}

		public void ProofPager()
		{
			if (mGridView != null)
			{
				if (mGridView.PageCount <= 1)
				{
					lbtnNext.Visible = false;
					lbtnPrevious.Visible = false;
					lbtnPrevious10.Visible = false;
					lbtnNext10.Visible = false;
					Repeater1.Visible = false;
				}
				else
				{
					if (mGridView.PageIndex < mGridView.PageCount - 1)
					{
						lbtnNext.Visible = true;

						if (mGridView.PageCount > 9)
						{
							PageCount = 9;

							if (mGridView.PageIndex < mGridView.PageCount - 10)	//-1 für 0 basierten Index, -9 für Next10
							{
								string strPageIndex = mGridView.PageIndex.ToString();

								string sPage = strPageIndex.Substring(0, 1);
								sPage += "0";
								PagerStart = Convert.ToInt16(sPage);

								lbtnNext10.Visible = true;
							}
						}
						else
						{
							PageCount = mGridView.PageCount - 1;

							lbtnNext10.Visible = false;
							lbtnPrevious10.Visible = false;
						}
					}
					else
					{
						lbtnNext.Visible = false;
					}

					if (mGridView.PageIndex > 0)
					{
						lbtnPrevious.Visible = true;

						if (mGridView.PageCount > 9)
						{
							PageCount = 9;

							if (mGridView.PageIndex > 9)
							{
								string strPageIndex = mGridView.PageIndex.ToString();

								string sPage = strPageIndex.Substring(0, 1);
								sPage += "0";
								PagerStart = Convert.ToInt16(sPage);

								lbtnPrevious10.Visible = true;
							}
							else
							{
								PagerStart = 0;
							}
						}
						else
						{
							PageCount = mGridView.PageCount - 1;

							lbtnNext10.Visible = false;
							lbtnPrevious10.Visible = false;
						}
					}
					else
					{
						lbtnPrevious.Visible = false;
					}										
				}			
			}

			if (mDataGrid != null)
			{
				if (mDataGrid.PageCount <= 1)
				{
					lbtnNext.Visible = false;
					lbtnPrevious.Visible = false;
					lbtnPrevious10.Visible = false;
					lbtnNext10.Visible = false;
					Repeater1.Visible = false;					
				}
				else
				{
					if (mDataGrid.CurrentPageIndex < mDataGrid.PageCount - 1)
					{
						lbtnNext.Visible = true;

						if (mDataGrid.PageCount > 9)
						{
							PageCount = 9;

							if (mDataGrid.CurrentPageIndex < mDataGrid.PageCount - 10)	//-1 für 0 basierten Index, -9 für Next10
							{
								string strPageIndex = mDataGrid.CurrentPageIndex.ToString();

								string sPage = strPageIndex.Substring(0, 1);
								sPage += "0";
								PagerStart = Convert.ToInt16(sPage);

								lbtnNext10.Visible = true;
							}
						}
						else
						{
							PageCount = mDataGrid.PageCount - 1;

							lbtnNext10.Visible = false;
							lbtnPrevious10.Visible = false;
						}
					}
					else 
					{ 
						lbtnNext.Visible = false; 
					}

					if (mDataGrid.CurrentPageIndex > 0)
					{
						lbtnPrevious.Visible = true;

						if (mDataGrid.PageCount > 9)
						{
							PageCount = 9;

							if (mDataGrid.CurrentPageIndex > 9)
							{
								string strPageIndex = mDataGrid.CurrentPageIndex.ToString();
								
								string sPage = strPageIndex.Substring(0, 1);
								sPage += "0";
								PagerStart = Convert.ToInt16(sPage);

								lbtnPrevious10.Visible = true;
							}
							else 
							{ 
								PagerStart = 0;
							}
						}
						else
						{
							PageCount = mDataGrid.PageCount - 1;

							lbtnNext10.Visible = false;
							lbtnPrevious10.Visible = false;
						}		
					}
					else
					{
						lbtnPrevious.Visible = false;
					}										
				}
				
				// Repeater füllen
				int i;
				DataTable LinkTable = new DataTable();

				LinkTable.Columns.Add("Index", Type.GetType("System.String"));
				LinkTable.Columns.Add("Page", Type.GetType("System.String"));

				for (i = 0; i >= PagerStart && i <= PageCount; i++) //Rows mit Index und Page hinzufügen
				{
					LinkTable.Rows.Add(new int[]{i,i+1});
				}

				Repeater1.DataSource = LinkTable;
				Repeater1.DataBind();
				Repeater1.Visible = true;
			}
		}

		protected void lbtnPrevious_Click(object sender, EventArgs e)
		{
			if (mDataGrid != null)
			{
                mDataGrid.CurrentPageIndex --;
                PagerChanged(this, mDataGrid.CurrentPageIndex);
			}
			else if (mGridView != null)
			{
                mGridView.PageIndex --;
                PagerChanged(this, mGridView.PageIndex);
			}
			else
			{
				throw new Exception("GridNavigation: noch nicht initialisiert");
			}
           
		}

		protected void lbtnPrevious10_Click(object sender, EventArgs e)
		{
            int tmpIntPageintex;
            //DataGridPageChangedEventArgs EventArg;

            if (mDataGrid != null)
            {
                tmpIntPageintex = mDataGrid.CurrentPageIndex - 9;
                if ((tmpIntPageintex).ToString().Length > 1)
                {
                    string sPage = (tmpIntPageintex).ToString().Substring(0, 1);
                    sPage += "0";
                    PagerStart = Convert.ToInt16(sPage);
                }
                tmpIntPageintex = PagerStart;
                //EventArg = new DataGridPageChangedEventArgs(mDataGrid, tmpIntPageintex);
                
            }
            else if (mGridView != null)
            {
                tmpIntPageintex = mGridView.PageIndex - 9;
                if ((tmpIntPageintex).ToString().Length > 1)
                {
                    string sPage = (tmpIntPageintex).ToString().Substring(0, 1);
                    sPage += "0";
                    PagerStart = Convert.ToInt16(sPage);
                }
                tmpIntPageintex = PagerStart;
                //EventArg = new DataGridPageChangedEventArgs(mGridView, tmpIntPageintex);
                
            }
            else
            {
                throw new Exception("GridNavigation: noch nicht initialisiert");
            }
            PagerChanged(this, tmpIntPageintex);
		}

		protected void lbtnNext10_Click(object sender, EventArgs e)
		{
            int tmpIntPageindex;
            //DataGridPageChangedEventArgs EventArg;

            if (mDataGrid != null)
            {
                tmpIntPageindex = mDataGrid.CurrentPageIndex + 9;
                if ((tmpIntPageindex + 1).ToString().Length > 1)
                {
                    String sPage = (tmpIntPageindex + 1).ToString().Substring(0, 1);
                    sPage += "0";
                    PagerStart = Convert.ToInt16(sPage);
                }
                tmpIntPageindex = PagerStart;
                //EventArg = new DataGridPageChangedEventArgs(mDataGrid, tmpIntPageintex);
            }
            else if (mGridView != null)
            {
                tmpIntPageindex = mGridView.PageIndex + 9;
                if ((tmpIntPageindex + 1).ToString().Length > 1)
                {
                    string sPage = (tmpIntPageindex + 1).ToString().Substring(0, 1);
                    sPage += "0";
                    PagerStart = Convert.ToInt16(sPage);
                }
                tmpIntPageindex = PagerStart;
                //EventArg = new DataGridPageChangedEventArgs(mGridView, tmpIntPageintex);                
            }
            else
            {
                throw new Exception("GridNavigation: noch nicht initialisiert");
            }
            PagerChanged(this, tmpIntPageindex);
		}

		protected void lbtnNext_Click(object sender, EventArgs e)
		{
            if (mDataGrid != null)
            {
                mDataGrid.CurrentPageIndex ++;
                PagerChanged(this, mDataGrid.CurrentPageIndex);
            }
            else if (mGridView != null)
            {
                mGridView.PageIndex ++;
                PagerChanged(this, mGridView.PageIndex);
            }
            else
            {
                throw new Exception("GridNavigation: noch nicht initialisiert");
            }
		}
        
		protected void ddlPageSize_SelectedIndexChanged(object sender, EventArgs e)
		{
			if (mDataGrid != null)
			{
				mDataGrid.PageSize = Convert.ToInt16(ddlPageSize.SelectedValue);
			}
			else if (mGridView != null)
			{
				mGridView.PageSize = Convert.ToInt16(ddlPageSize.SelectedValue);
			}
			else
			{
				throw new Exception("GridNavigation: noch nicht initialisiert");
			}
			PageSizeChanged();
		}

        //public void lbtnPage_PageIndexChanging(object sender, EventArgs e)
        //{
        //    LinkButton lbtn = (LinkButton)sender;
        //    if (mDataGrid != null)
        //    {
        //        PagerChanged(this,Convert.ToInt32(lbtn.CommandArgument) );//new DataGridPageChangedEventArgs(mDataGrid, Convert.ToInt32(lbtn.CommandArgument))
        //    }
        //    else if (mGridView != null)
        //    {
        //        PagerChanged(this, Convert.ToInt32(lbtn.CommandArgument)); //new DataGridPageChangedEventArgs(mGridView, Convert.ToInt32(lbtn.CommandArgument))
        //    }
        //}

		private void Page_PreRender(object sender, EventArgs e)
		{
			//ProofPager();

			lblAnzahl.Visible = true;

			if (mDataGrid != null)
			{
				if (mDataGrid.DataSource is DataView)
				{
					lblAnzahl.Text = "Gesamtanzahl: " + ((DataView)mDataGrid.DataSource).Count;
				}
				else if (mDataGrid.DataSource is DataTable)
				{
					lblAnzahl.Text = "Gesamtanzahl: " + ((DataTable)mDataGrid.DataSource).Rows.Count;
				}
				else
				{
					lblAnzahl.Visible = false;
				}
			}
			else if (mGridView != null)
			{
				if (mGridView.DataSource is DataView)
				{
					lblAnzahl.Text = "Gesamtanzahl: " + ((DataView)mGridView.DataSource).Count;
				}
				else if (mGridView.DataSource is DataTable)
				{
					lblAnzahl.Text = "Gesamtanzahl: " + ((DataTable)mGridView.DataSource).Rows.Count;
				}
				else
				{
					lblAnzahl.Visible = false;
				}
			}
			else
			{
				throw new Exception("GridNavigation: noch nicht initialisiert");
			}
		}
	}
}
